pyomo/contrib/appsi/solvers/gurobi.py

```diff
@@ -257,42 +257,46 @@ def __init__(self, only_child_vars=True):
         self._last_results_object: Optional[GurobiResults] = None
 
     def available(self):
-        if self._available is None:
-            self._check_license()
-        return self._available
+        if not gurobipy_available:  # this triggers the deferred import
+            return self.Availability.NotFound
+        elif self._available == self.Availability.BadVersion:
+            return self.Availability.BadVersion
+        else:
+            return self._check_license()
 
-    @classmethod
-    def _check_license(cls):
+    def _check_license(self):
+        avail = False
         try:
             # Gurobipy writes out license file information when creating
             # the environment
             with capture_output(capture_fd=True):
                 m = gurobipy.Model()
-                m.dispose()
-        except ImportError:
-            # Triggered if this is the first time the deferred import of
-            # gurobipy is resolved. _import_gurobipy will have already
-            # set _available appropriately.
-            return
+            if self._solver_model is None:
+                self._solver_model = m
```

Comment:
This looks like the solver interface will create (and perpetually hold on to) a Gurobi model -- I suspect to grab the license and keep others from "stealing" it from us.  Can you document somewhere (preferably obvious for users) the strategy that we are using for managing Gurobi licenses (and in particular what a user should do to release their license)?

Reply:
Done!